packages/console/src/hooks/use-connector-in-use.ts

```diff
@@ -1,41 +1,41 @@
-import type { SignInExperience } from '@logto/schemas';
+import type { ConnectorResponse, SignInExperience } from '@logto/schemas';
 import { SignInIdentifier, ConnectorType } from '@logto/schemas';
+import { useCallback } from 'react';
 import useSWR from 'swr';
 
 import type { RequestError } from './use-api';
 
-const useConnectorInUse = (type?: ConnectorType, target?: string): boolean | undefined => {
-  const { data } = useSWR<SignInExperience, RequestError>(target && type && '/api/sign-in-exp');
-
-  if (!data) {
-    return;
-  }
-
-  if (type === ConnectorType.Email) {
-    return (
-      data.signIn.methods.some(
-        ({ identifier, verificationCode }) =>
-          verificationCode && identifier === SignInIdentifier.Email
-      ) ||
-      (data.signUp.identifiers.includes(SignInIdentifier.Email) && data.signUp.verify)
-    );
-  }
-
-  if (type === ConnectorType.Sms) {
-    return (
-      data.signIn.methods.some(
-        ({ identifier, verificationCode }) =>
-          verificationCode && identifier === SignInIdentifier.Sms
-      ) ||
-      (data.signUp.identifiers.includes(SignInIdentifier.Sms) && data.signUp.verify)
-    );
-  }
-
-  if (!target) {
-    return;
-  }
-
-  return data.socialSignInConnectorTargets.includes(target);
+const useConnectorInUse = () => {
+  const { data } = useSWR<SignInExperience, RequestError>('/api/sign-in-exp');
+
+  const isConnectorInUse = useCallback(
+    (connector?: ConnectorResponse) => {
+      if (!connector || !data) {
+        return false;
+      }
+
+      const { type, target } = connector;
+
+      if (type === ConnectorType.Social) {
+        return data.socialSignInConnectorTargets.includes(target);
+      }
+
+      const relatedIdentifier =
+        type === ConnectorType.Email ? SignInIdentifier.Email : SignInIdentifier.Sms;
+
+      return (
+        data.signIn.methods.some(
+          ({ identifier, verificationCode }) => verificationCode && identifier === relatedIdentifier
+        ) ||
+        (data.signUp.identifiers.includes(relatedIdentifier) && data.signUp.verify)
+      );
```

Comment:
I'd recommend refactor this logic, and split it into smaller pieces. For example:
```ts
const isSignInIdentifier = ...
const isSignUpIdentifier = ...

return isSignInIdentifier || isSignUpIdentifier;
```

Reply:
updated in https://github.com/logto-io/logto/pull/2813